l10n_it_edi_related_document_type/models/account_move.py

```diff
@@ -0,0 +1,114 @@
+from odoo import fields, models
+
+from odoo.addons.l10n_it_edi.models.account_move import get_text
+
+RELATED_DOCUMENT_TYPES = {
+    "order": "DatiOrdineAcquisto",
+    "contract": "DatiContratto",
+    "agreement": "DatiConvenzione",
+    "reception": "DatiRicezione",
+    "invoice": "DatiFattureCollegate",
+}
+
+
+class AccountMoveRelatedDocumentType(models.Model):
+    _name = "account.move.related_document_type"
+    _description = "E-invoice Related Document Type"
+
+    type = fields.Selection(
+        [
+            ("order", "Order"),
+            ("contract", "Contract"),
+            ("agreement", "Agreement"),
+            ("reception", "Reception"),
+            ("invoice", "Related Invoice"),
+        ],
+        "Document Type",
+        required=True,
+    )
+    name = fields.Char("Document ID", size=20, required=True)
+    lineRef = fields.Integer("Line Ref.")
+    # invoice_line_id = fields.Many2one(
+    #     "account.move.line",
+    #     "Related Invoice Line",
+    #     ondelete="cascade",
+    #     index=True,
+    #     readonly=True,
+    # )
+    invoice_id = fields.Many2one(
+        "account.move",
+        "Related Invoice",
+        ondelete="cascade",
+        index=True,
+        readonly=True,
+    )
+    date = fields.Date()
+    numitem = fields.Char("Item Num.", size=20)
+    code = fields.Char("Order Agreement Code", size=100)
+    cig = fields.Char("CIG Code", size=15)
+    cup = fields.Char("CUP Code", size=15)
+
+    # @api.model_create_multi
+    # def create(self, vals_list):
+    #     line_obj = self.env["account.move.line"]
+    #     for vals in vals_list:
+    #         if vals.get("invoice_line_id"):
+    #             line = line_obj.browse(vals["invoice_line_id"])
+    #             vals["lineRef"] = line.sequence
+    #     return super().create(vals_list)
+
+    # def setlineRef(self, n):
+    #     self.lineRef = n
+
+
+class AccountMove(models.Model):
+    # _position = ['2.1', '2.2', '2.3', '2.4', '2.5']
+    _inherit = "account.move"
+
+    related_documents = fields.One2many(
```

Comment:
```suggestion
    related_document_ids = fields.One2many(
```
vedi https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#fields:
> One2Many and Many2Many fields should always have _ids as suffix (example: sale_order_line_ids)

Reply:
:+1: